### ISSUE TITLE
`org.jctools.util.UnsafeAccess` invokes terminally deprecated method `sun.misc.Unsafe::objectFieldOffset`

### ISSUE THREAD

**Issue by zakkak** [JCTools/JCTools#395]
Using JCTools with JDK >=24 (see https://openjdk.org/jeps/498) results in the following warning:

```
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by org.jctools.util.UnsafeAccess (file:/home/zakkak/.m2/repository/org/jctools/jctools-core/4.0.5/jctools-core-4.0.5.jar)
WARNING: Please consider reporting this to the maintainers of class org.jctools.util.UnsafeAccess
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
```

**Comment by nitsanw** [comment]
Yes... This is expected. Users can use atomic queues as an Unsafe free alternative (where possible). We should look into an Unsafe free version of JCTools, which is a lot of work to convert the code, retest and re-benchmark.

**Comment by tmccombs** [comment]
Could this be replaced by the new memory api?

**Comment by franz1981** [comment]
> Could this be replaced by the new memory api?

VarHandle should just work: https://foojay.io/today/unsafe-is-finally-going-away-embracing-safer-memory-access-with-jep-471/
but as said by @nitsanw ; it's not just matter of replacing it (although it's a risk/cost per se) - but we need to evaluate the performance impact and eventually which mitigations (if any), to fix it.
It's true as well as the world move past `Unsafe` there won't be any alternative, but we are still approaching this world and not there yet (think about Netty 4.2 which is has java 8 as base JDK version).
